### ISSUE TITLE
Freezing picture when switching mode on hammerhead

### ISSUE THREAD

**Issue by nfsprodriver** [ubports/camera-app#18]
I don't know, whether this is realted to camera app, this topic is also discussed in the ubports-touch repo.

**Comment by dark-eye** [comment]
hey @nfsprovider 
I'm asumming the mode switching is between video/stills modes , does the viewport aspect ratio change between the mode switching? As i noticed that i get some small freezes when i change aspect ratio and then switch between the modes.

Also does it a complete freeze or just a short one?

Thanks for reporting :)

**Comment by Flohack74** [comment]
I can confirm that I cannot switch from photo to video mode. I get a black window, but it starts working when swiping for switching to another app, but then just selecting the camera again ;)

**Comment by dark-eye** [comment]
Since i don't have  Hmmerhead I can't really investigate this issue if any one else has the time it will be a great  help.

Thanks

**Comment by dark-eye** [comment]
Since refocusing  on the app solves the issue It seems that the issue is  either in the resolution update or in the  camera state please install the version below  and see if the issue is fixed : 
	
[com.ubuntu.camera_3.0.1.759_armhf.click](https://my.owndrive.com/index.php/s/2hoXuKuiuPLPNKp)

**Comment by GI9VANNI** [comment]
I do not know if it's the same problem, but when I try to record a video with the Nexus 5, the screen freezes and I have to close the app.

**Comment by advocatux** [comment]
Switching between photo and video mode works fine for me with v. 3.0.1(31/12/17) but it still freezes when switching resolution mode.

**Comment by dark-eye** [comment]
great were advancing;) @advoxatux any chance you can send/link the log of a resultion change freeze?

**Comment by advocatux** [comment]
Sure, which are the relevant logs I should send?

**Comment by dark-eye** [comment]
@advocatux  If the application freezes just the camera app (v3.0.1) should suffice

**Comment by GI9VANNI** [comment]
I did some tests with the latest update (3.0.1) and I confirm that when I try to record a video with the Nexus 5, the screen freezes and I have to close the app.
However, as @Flohack74 says, if I swipe for switching to another app and then I select the camera, the recording works! Playback of recordings is bad anyway, they are seen in green and purple.
I hope this can help you! @dark-eye @advocatux

**Comment by advocatux** [comment]
@GI9VANNI I record video without any problem in Nexus 5 with the latest camera app version, and without needing to do any trick.
You can playback the video recorded with UTmedia. @mariogrip explains that bug, and why it isn't fixed yet, here https://github.com/ubports/ubuntu-touch/issues/7

@dark-eye can you point me out the exact path to the relevant log/s? Thank you

**Comment by GI9VANNI** [comment]
@advocatux We need to understand why it works without problems for you and not for me. There may be some small difference between the two devices or something else.

**Comment by advocatux** [comment]
@GI9VANNI yes of course. I wanted to stand out that this is not a case of "it never works".

**Comment by GI9VANNI** [comment]
@advocatux O.K.
